fda/fdata_boxplot.py

```diff
@@ -0,0 +1,627 @@
+                if fig and ax are None.
+
+         Returns:
+            fig (figure object): figure object in which the graphs are plotted in case ax is None.
+            ax (axes object): axes in which the graphs are plotted.
+
+        """
+
+        fig, ax = self.fdatagrid.generic_plotting_checks(fig, ax, nrows,
+                                                         ncols)
+        tones = np.linspace(0.1, 1.0, len(self._prob) + 1, endpoint=False)[1:]
+        color = self.colormap(tones)
+
+        if self.fullout:
+            var_zorder = 4
+        else:
+            var_zorder = 1
+
+        for m in range(self.fdatagrid.ndim_image):
+
+            # outliers
+            for j in list(range(self.fdatagrid.nsamples)):
+                if self.outliers[m, j]:
+                    ax[m].plot(self.fdatagrid.sample_points[0],
+                               self.fdatagrid.data_matrix[j, :, m],
+                               color=self.outliercol,
+                               linestyle='--', zorder=1)
+
+            for i in range(len(self._prob)):
+                # central regions
+                ax[m].fill_between(self.fdatagrid.sample_points[0],
+                                   self.central_regions[
+                                       m * len(self._prob) + i, 0],
+                                   self.central_regions[
+                                       m * len(self._prob) + i, 1],
+                                   facecolor=color[i], zorder=var_zorder)
+
+            # outlying envelope
+            ax[m].plot(self.fdatagrid.sample_points[0],
+                       self.outlying_envelope[m, 0],
+                       self.fdatagrid.sample_points[0],
+                       self.outlying_envelope[m, 1], color=self.barcol,
+                       zorder=4)
+
+            # central envelope
+            ax[m].plot(self.fdatagrid.sample_points[0],
+                       self.central_envelope[m, 0],
+                       self.fdatagrid.sample_points[0],
+                       self.central_envelope[m, 1], color=self.barcol,
+                       zorder=4)
+
+            # vertical lines
+            index = math.ceil(self.fdatagrid.ncol / 2)
+            x = self.fdatagrid.sample_points[0][index]
+            ax[m].plot([x, x], [self.outlying_envelope[m, 0][index],
+                                self.central_envelope[m, 0][index]],
+                       color=self.barcol,
+                       zorder=4)
+            ax[m].plot([x, x], [self.outlying_envelope[m, 1][index],
+                                self.central_envelope[m, 1][index]],
+                       color=self.barcol, zorder=4)
+
+            # median sample
+            ax[m].plot(self.fdatagrid.sample_points[0], self.median[m],
+                       color=self.mediancol, zorder=5)
+
+        self.fdatagrid.set_labels(fig, ax)
+
+        return fig, ax
+
+    def _repr_svg_(self):
```

Comment:
You should test this method in the examples, by using `fdboxplot` as the last line, instead of plotting explicitly.

Reply:
I am not very familiar with the Jupyter Notebooks generated by Sphinx but I noticed that only the outputs consisting in images of plotting methods are inserted. 
``` fdboxplot``` default representation is an image but it is not shown, maybe it is related to the fact of using StringIO and it is recognized as an string. I will investigate more.

Reply:
If you go to the `Examples which don’t produce image output` section in https://sphinx-gallery.readthedocs.io/en/latest/auto_examples/index.html, you can see that unless the `print` method is used no output is shown, and the print function calls the _str_ method, so this functionality cannot be shown in the examples :( I left it with the plot method to show the images and introduced a comment at the end.

Reply:
Have you tried the `display` function provided by IPython?

Reply:
The ´display´ function does not work neither, I have come to https://github.com/sphinx-gallery/sphinx-gallery/issues/396 but I do not understand it very much, maybe you could clear up something. Also maybe it can be resolved using ´Custom Scrapers´ explained in https://sphinx-gallery.readthedocs.io/en/latest/configuration.html.

Reply:
Ok, leave it like that for now.